autopts/pybtp/btp/bap.py

```
@@ -130,12 +130,13 @@ def bap_send(ase_id, data_ba, bd_addr_type=None, bd_addr=None):
 def bap_broadcast_source_setup(
         streams_per_subgroup, subgroups, coding_format, vid, cid,
         codec_ltvs, sdu_interval, framing, max_sdu, retransmission_number,
-        max_transport_latency, presentation_delay):
+        max_transport_latency, presentation_delay, broadcast_id=0):
```

Given that you've  positioned `broadcast_id` as the first field in the packet, I suggest to also keep it as the first argument to this function

---

> Wouldn't it be better to keep it at the end of the packet as well? That way it doesn't affect those that doesn't look for it in the packet? Meaning they only put the first bytes up to the broadcast ID into a struct

Perhaps. My assumption would be that, since it's a static size command, that IUTs would do a `==` size check before parsing it (that's what Zephyr does at least), which would break regardless. 

But you are right that there's a higher chance (even if little) that it will be less disruptive if put at last (that's also the approach the BT SIG uses when releasing new versions of existing commands and events). 

I'd like to get input from @sjanc on this

---

I see now why I put it at the start of the packet, there is an LTV at the end with an unknown size, meaning you would need to implement a handler for it, similar to `btp_gap_start_advertising_cmd`

---

> I see now why I put it at the start of the packet, there is an LTV at the end with an unknown size, meaning you would need to implement a handler for it, similar to `btp_gap_start_advertising_cmd`

Ah, yeah that makes sense. Then it cannot be at the very end at least

---

changing  command is always a problem unfortunately,  sometimes we were able to workaround this with extra commands that set a specific state in IUT (which could be used here),   very seldom we just broke BTP (but this was done only for newly added commands afair where only single [kown] user were present)

alternative would be to just add v2 command (with new opcode) and use it (if IUT claimed support for it)  [2]
yet, with this approach we should also be conservative since BTP supports only 127 commands per service...   this might not feel as big issue but... mesh services already have 100+ commands.. ;-)

Since this is likely to start becoming more of an issue, one solution could be to start thinking about versioning BTP,  we knew it would a thing at some point [1].  I think versioning (probably might be an easiest and simplest approach,   when IUT is using too old version of BTP we could simply fail early with dedicated error.  This would at least give very clear hint to update IUT BTP implementation instead of odd regressions in (random) tests..

[1] https://github.com/auto-pts/auto-pts/issues/337
[2] https://github.com/auto-pts/auto-pts/issues/1251